test/cbmc/proofs/decodeFileBlock_Mqtt/decodeFileBlock_Mqtt_harness.c

```diff
@@ -0,0 +1,53 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * @file decodeFileBlock_Mqtt_harness.c
+ * @brief Implements the proof harness for decodeFileBlock_Mqtt function.
+ */
+/* Include headers for mqtt interface. */
+#include "ota_mqtt_private.h"
+
+bool OTA_CBOR_Decode_GetStreamResponseMessage( const uint8_t * pMessageBuffer,
+                                               size_t messageSize,
+                                               int32_t * pFileId,
+                                               int32_t * pBlockId,
+                                               int32_t * pBlockSize,
+                                               uint8_t ** pPayload,
+                                               size_t * pPayloadSize )
+{
+    bool status;
+
+    /* If any of the arguments to the function are not initialized
+     * correctly then the functions should return FALSE. */
+    if( ( pFileId == NULL ) ||
+        ( pBlockId == NULL ) ||
+        ( pBlockSize == NULL ) ||
+        ( pPayload == NULL ) ||
+        ( pPayloadSize == NULL ) ||
+        ( pMessageBuffer == NULL ) )
+    {
+        status = false;
+    }
+
+    return status;
+}
```

Comment:
Do we need to mock this function instead of using it directly? If we can use it directly, that would model the behavior of the code better.

In this case this function usually mallocs some memory and then the caller needs to free it. So I think it could make sense to use the original function.

Reply:
IMO it's not required for to call the function directly. Using a stub saves time in reproducing a similar result and is computationally efficient. The called function is also checked for memory safety.

Reply:
Okay if the called function is tested as well then that's okay.